File: s01_Basic/SimpleCalculator/app/src/main/java/com/omeryildizce/simplecalculator/MainActivity.java
package com.omeryildizce.simplecalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    TextView resultText;
    EditText number1Text;
    EditText number2Text;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        creator();
    }

    private void creator() {
        resultText = findViewById(R.id.resultText);
        number1Text = findViewById(R.id.number1Text);
        number2Text = findViewById(R.id.number2Text);

    }

    public void sum(View view) {
        if (number1Text.getText().toString().matches("") || number2Text.getText().toString().matches("")) {
            resultText.setText(getResources().getString(R.string.enter_number));
        } else {

            int number1 = Integer.parseInt(number1Text.getText().toString());
            int number2 = Integer.parseInt(number2Text.getText().toString());
            int result = number1 + number2;
            String text = getResources().getString(R.string.result) + " " + result;
            resultText.setText(text);
        }
    }

    public void deduct(View view) {
        if (number1Text.getText().toString().matches("") || number2Text.getText().toString().matches("")) {
            resultText.setText(getResources().getString(R.string.enter_number));
        } else {

            int number1 = Integer.parseInt(number1Text.getText().toString());
            int number2 = Integer.parseInt(number2Text.getText().toString());
            int result = number1 - number2;
            String text = getResources().getString(R.string.result) + " " + result;
            resultText.setText(text);
        }
    }

    public void multiply(View view) {
        if (number1Text.getText().toString().matches("") || number2Text.getText().toString().matches("")) {
            resultText.setText(getResources().getString(R.string.enter_number));
        } else {

            int number1 = Integer.parseInt(number1Text.getText().toString());
            int number2 = Integer.parseInt(number2Text.getText().toString());
            int result = number1 * number2;
            String text = getResources().getString(R.string.result) + " " + result;
            resultText.setText(text);
        }
    }

    public void divide(View view) {
        if (number1Text.getText().toString().matches("") || number2Text.getText().toString().matches("")) {
            resultText.setText(getResources().getString(R.string.enter_number));
        } else {

            int number1 = Integer.parseInt(number1Text.getText().toString());
            int number2 = Integer.parseInt(number2Text.getText().toString());
            int result = number1 / number2;
            String text = getResources().getString(R.string.result) + " " + result;
            resultText.setText(text);
        }
    }

    public void mod(View view) {
        if (number1Text.getText().toString().matches("") || number2Text.getText().toString().matches("")) {
            resultText.setText(getResources().getString(R.string.enter_number));
        } else {

            int number1 = Integer.parseInt(number1Text.getText().toString());
            int number2 = Integer.parseInt(number2Text.getText().toString());
            int result = number1 % number2;
            String text = getResources().getString(R.string.result) + " " + result;
            resultText.setText(text);
        }
    }
}

File: s15_KotlinLearning/SimpleCalculator/app/src/main/java/com/omeryildizce/simplecalculator/MainActivity.kt
package com.omeryildizce.simplecalculator

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.omeryildizce.simplecalculator.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private var number1 = 0
    private var number2 = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)



    }

    private fun initalize( ) {
        number1 = binding.editTextNumber1.text.toString().toIntOrNull() ?: 0
        number2 = binding.editTextNumber2.text.toString().toIntOrNull() ?: 0

    }
    private fun writeResult(result:Int){
        binding.textViewResult.setText("Result: $result")
    }

    fun mySum(view: View) {
        initalize()
        writeResult(number1 + number2)

    }
    fun mySub(view: View) {
        initalize()
        writeResult( number1 - number2)
    }
    fun myMultiply(view: View) {
        initalize()
        writeResult( number1 * number2)
    }
    fun myDvide(view: View) {
        initalize()
        if (number2!=0)
            writeResult( (number1 / number2).toInt())
        else Toast.makeText(this, "Number 2 cannot be 0 or null!", Toast.LENGTH_SHORT).show()
    }
}